spec/controllers/advice_controller_spec.rb

```diff
@@ -136,7 +214,59 @@
         expect(result).to render_template(:edit_advice)
       end
     end
+    
+  describe '#save_advice' do
```

Comment:
From here to line 269 are duplicates.  And there is an extra end in there lower as well I believe.